pkgs/data/misc/media-types/default.nix

```diff
@@ -0,0 +1,40 @@
+{ lib, stdenv, fetchFromGitLab }:
+
+stdenv.mkDerivation rec {
+  pname = "media-types";
+  version = "8.0.0";
+
+  src = fetchFromGitLab {
+    domain = "salsa.debian.org";
+    owner = "debian";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-dl6XKxOF9YT9n/iSvZj1juTqSMaZuOF5i7Eb/VZZDFY";
+  };
+
+  installPhase = ''
+    install -D -m0444 mime.types $out/etc/mime.types
+    install -D -m0444 debian/bug-presubj $out/share/bug/media-types/presubj
+    install -D -m0444 debian/changelog $out/share/doc/media-types/changelog
+    install -D -m0444 debian/copyright $out/share/doc/media-types/copyright
+
+    # Generate nginx mime-types
+    mkdir -p  $out/etc/nginx
+    echo "types {" > $out/etc/nginx/nginx-mime.types
+    cat mime.types | sed -e '/^#.*/d' -e '/^$/d' -e 's/.*/\ \ &;/' -ne '/[^[:space:]][[:space:]]\{1,\}[^[:space:]]/p' >> $out/etc/nginx/nginx-mime.types
+    echo "}" >> $out/etc/nginx/nginx-mime.types
```

Comment:
You could probably reuse `mailcap`'s [`generate-nginx-mimetypes.sh`](https://pagure.io/mailcap/blob/master/f/generate-nginx-mimetypes.sh) script for this.

Reply:
There will be an unnecessary dependency on the mailcap package.
If there are changes in the `mailcap` package, then a mass rebuild of the packages in the system will follow.
And here the script is a little different.